PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs

```diff
@@ -45,16 +45,154 @@ public List<Prescription> GetPrescriptionsForPatient(int idPatient)
+            }
+            else
+            {
+                return searchForDoctorAdvanced(prescriptions, dto.First);
+            }
+        }
+
+        private List<Prescription> searchForDoctorAdvanced(List<Prescription> prescriptions, String searchField)
+        {
+            if (!searchField.Equals(""))
+            {
+                prescriptions = prescriptions.FindAll(prescription => prescription.Doctor.firstName.Contains(searchField) || prescription.Doctor.secondName.Contains(searchField) || prescription.Doctor.DoctorFullName().Contains(searchField));
+            }
+
+            return prescriptions;
+        }
+
+        private List<Prescription> searchForUsedAdvanced(List<Prescription> prescriptions, String searchField)
+        {
+            if (!searchField.Equals(""))
+            {
+                prescriptions = prescriptions.FindAll(prescription => prescription.isUsed.ToString().Contains(searchField));
+            }
+
+            return prescriptions;
+        }
+
+        private List<Prescription> searchForCommentsAdvanced(List<Prescription> prescriptions, String searchField)
+        {
+            if (!searchField.Equals(""))
+            {
+                prescriptions = prescriptions.FindAll(prescription => prescription.comment.Contains(searchField));
+            }
+
+            return prescriptions;
+        }
+
+        private List<Prescription> searchForMedicinesAdvanced(List<Prescription> prescriptions, String searchField)
+        {
+            if (!searchField.Equals(""))
+            {
+                prescriptions = prescriptions.Where(prescription => prescription.Medicines.Any(medicine => medicine.name.Contains(searchField))).ToList();
+            }
 
-            prescriptions = searchForMedicines(prescriptions, prescriptionSearchDto);
-       
             return prescriptions;
+        }
 
+        /// <summary> This method is getting list of filtered <c>Prescription</c> of logged patient by parameter <c>Doctor</c>. </summary>
+        /// /// <param name="prescriptions"><c>prescriptions</c> is List of presciptions that matches search fields.
+        /// </param>
+        /// /// <param name="prescriptionSearchDto"><c>prescriptionSearchDto</c> is Data Transfer Object of a <c>Prescription</c> that is beomg used to filter precriptions.
+        /// </param>
+        /// <returns> List of filtered patient prescriptions. </returns>
+        private List<Prescription> searchForDoctor(List<Prescription> prescriptions, PrescriptionSearchDto prescriptionSearchDto)
+        {
+            if (!prescriptionSearchDto.Doctor.Equals(""))
+            {
+                prescriptions = prescriptions.FindAll(prescription => prescription.Doctor.firstName.Contains(prescriptionSearchDto.Doctor) || prescription.Doctor.secondName.Contains(prescriptionSearchDto.Doctor) || prescription.Doctor.DoctorFullName().Contains(prescriptionSearchDto.Doctor));
+            }
+
+            return prescriptions;
```

Comment:
For each condition where it is checked whether a string is empty use the method implemented for that check within the utility methods class .